misk-crypto/src/main/kotlin/misk/crypto/CryptoModule.kt

```diff
@@ -49,76 +53,112 @@ class CryptoModule(
+      check(duplicateNames.isEmpty()) {
+        "Found duplicate keys: [$duplicateNames]"
+      }
     }
 
-    config.keys.forEach { key ->
-      when (key.key_type) {
-        KeyType.AEAD -> {
-          bind<Aead>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(AeadEnvelopeProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-        }
-        KeyType.DAEAD -> {
-          bind<DeterministicAead>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(DeterministicAeadProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-        }
-        KeyType.MAC -> {
-          bind<Mac>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(MacProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-        }
-        KeyType.DIGITAL_SIGNATURE -> {
-          bind<PublicKeySign>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(DigitalSignatureSignerProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-          bind<PublicKeyVerify>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(DigitalSignatureVerifierProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-        }
-        KeyType.HYBRID_ENCRYPT -> {
-          bind<HybridEncrypt>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(HybridEncryptProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-        }
-        KeyType.HYBRID_ENCRYPT_DECRYPT -> {
-          bind<HybridDecrypt>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(HybridDecryptProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-          bind<HybridEncrypt>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(HybridEncryptProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-        }
-        KeyType.STREAMING_AEAD -> {
-          bind<StreamingAead>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(StreamingAeadProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-        }
-        KeyType.PGP_DECRYPT -> {
-          bind<PgpDecrypter>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(PgpDecrypterProvider(key, config.kms_uri))
-              .`in`(Singleton::class.java)
-        }
-        KeyType.PGP_ENCRYPT -> {
-          bind<PgpEncrypter>()
-              .annotatedWith(Names.named(key.key_name))
-              .toProvider(PgpEncrypterProvider(key))
-              .`in`(Singleton::class.java)
-        }
+    // Include all configured remote keys
+    config.external_data_keys?.let { external_data_keys ->
+      requireBinding<AmazonS3>()
```

Comment:
This crates an explicit dependency on an injected object. Normally, if it's not provided, this will fail at object use/injection site, where as in this case it will fail at the injector creation time.